### ISSUE TITLE
IndentationCheck: enum values with arguments is inconsistent and wrong

### ISSUE THREAD

**Issue by blerner** [checkstyle/checkstyle#5248]
/var/tmp $ cat YOUR_FILE.java
```java
public enum LambdasAndEnums {
  TEST("the lambda should be indented directly below this string",
       (x) -> x,
            "this string isn't indented correctly, nor is the number below",
    5)
}
```

/var/tmp $ cat config.xml
```xml
<!-- The relevant bit of configuration -->
        <module name="Indentation">
            <property name="basicOffset" value="2"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="2"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="4"/>
            <property name="arrayInitIndent" value="2"/>
            <message key="indentation.error"
                     value="`{0}` has incorrect indentation level {1}; its expected level should be {2}." />
        </module>
```
/var/tmp $ java -jar checkstyle-8.4-all.jar -c config.xml YOUR_FILE.java
```
`lambda arguments` has incorrect indentation level 7; its expected level should be 6.
```
---------------

I expect the line with the lambda to be just fine, since it's aligned with the argument above it.  I expect the second string and the number argument to generate indentation errors since they're clearly unaligned to anything.  But neither of these are the case.  Instead, if I outdent the lambda line by one character, so that its opening paren is (incorrectly) aligned with the opening paren of the `TEST` enum's constructor call, then it doesn't get an error.  And nothing I do seems to make the second string or number generate an error.

This inconsistency has bitten one of my students, whose actual use case was somewhat more complicated.  He was sufficiently confused that he thought the problem was the style checker can't handle nested lambdas (i.e., he had a curried function `(arg1) -> (arg2) -> (arg3) -> answer`), but it seems from this reduced case that the problem really lies with enum values with constructor arguments.

**Comment by romani** [comment]
http://checkstyle.sourceforge.net/config_misc.html#Indentation

strange that `forceStrictCondition` that is `false` by default does not considered. There should be no violation.

**Comment by blerner** [comment]
This has now bitten another one of my students, on the following code.  It has nothing to do with enums this time...  but nearly every line of this function gets complained about indentation.

```java
  private static <T extends Throwable> void assertThrows(IAction action,
                                                         Class<T> exceptionType,
                                                         Predicate<String> messageTester,
                                                         Function<String, String> messageTestFail) {
    assertThrows(
            action,
            /* STARTING WITH THIS LINE */
            (t) -> exceptionType.isInstance(t) && messageTester.test(t.getMessage()),
            (t) -> {
              if (!exceptionType.isInstance(t)) {
                return "Expected " + exceptionType.getName()
                        + " to be thrown, got " + t.getClass().getName();
              } else {
                return messageTestFail.apply(t.getMessage());
              }
            }, "Expected " + exceptionType.getName() + " to be thrown.");
            /* THROUGH TO HERE */
  }
```

**Comment by romani** [comment]
@blerner , can one of your student to fix the problem as some major work during education process ? :) Real life open source coding experience , to produce something valuable and reuse results right after work is done.

**Comment by blerner** [comment]
Not as part of our course, no -- it's an introductory course, they don't have the larger-scale software engineering skills yet to contribute to a project this scale.  We obviously do have a SwEng course, but it's taught somewhat differently (a semester-long project, rather than picking up external projects and contributing to them).

**Comment by romani** [comment]
We have a lot of tasks that could be semester long. Let me know if find some opportunities for students to do big projects in open source. We have a lot of indentation issues, fixing them all is long project that might result in rewriting Check completely.

**Comment by Abhishek-kumar09** [comment]
Fixed with #8719 
```
class S {
  private static <T extends Throwable> void assertThrows(IAction action,
                                                          Class<T> exceptionType,
                                                          Predicate<String> messageTester,
                                                          Function<String, String> messageTestFail) {
    assertThrows(
            action,
            /* STARTING WITH THIS LINE */
            (t) -> exceptionType.isInstance(t) && messageTester.test(t.getMessage()),
            (t) -> {
              if (!exceptionType.isInstance(t)) {
                return "Expected " + exceptionType.getName()
                        + " to be thrown, got " + t.getClass().getName();
              } else {
                return messageTestFail.apply(t.getMessage());
              }
            }, "Expected " + exceptionType.getName() + " to be thrown.");
            /* THROUGH TO HERE */
  }
}

public enum LambdasAndEnums {
  TEST("the lambda should be indented directly below this string",
       (x) -> x,
            "this string isn't indented correctly, nor is the number below",
    5)
}
```
config:
```
      <module name="Indentation">
             <property name="basicOffset" value="2"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="2"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="4"/>
            <property name="arrayInitIndent" value="2"/>
          </module>
```
output:
```
java -jar checkstyle-8.36.2-all.jar -c config.xml Test.java 
Starting audit...
Audit done.
```

**Comment by romani** [comment]
Fixed in scope of https://github.com/checkstyle/checkstyle/issues/3342
Released as 8.36.2
